### DISCUSSION TITLE
Lock Feature

### DISCUSSION THREAD

**Discussion by SpryzenR** [ScratchAddons/ScratchAddons#7342] [Ideas]
Hi guys I'm back for now, I'm trying to do well in school but if I get anymore ideas I'll let you know :)

### Lock Feature

Add a right click feature that locks things. In the settings, it'll give Scratchers two options for it to appear in (turn off and on)

-Paint Editor (Costumes and Background Editor)
-Sound Editor
-Coding Area

I saw this feature in Canva and thought it would be useful to implement into Scratch to decrease mistakes (create a poster design and add something random to get the feature).

**Comment by DNin01** [comment]
So basically a read-only mode! I could maybe see the use in this.

**Comment by DNin01** [comment]
I'll mention that a Nintendo Switch game that's more of a toolkit called Game Builder Garage has an option called "write protect" that you can turn on or off individually for every game you create, and it simply prevents you from editing the game until you turn it off. It's useful if you've finished a game and, I don't know, want to show it to your friends without the fear of someone messing anything up.

Just in case we want to draw any inspiration from that.

**Reply by SpryzenR** [reply]
Maybe adding on to my idea we can make it so specific costumes, sounds, backgrounds, sprites, or scripts are locked.
